### DISCUSSION TITLE
How to run an old runtime on a new node?

### DISCUSSION THREAD

**Discussion by timbrinded** [paritytech/zombienet#937] [Q&A]
What is the correct way to run a zombie network whereby a node is running a new client binary, but an old runtime?

When providing a `chain_spec_path` to a modified plain-spec, the runtime.system.code value is overriden as part of the build spec path.

This is a typical scenario as forward-compatible client software is released prior to networks being upgraded with new runtimes.

**Comment by pepoviola** [comment]
Hi @timbrinded, thanks for open this topic. I will check how we can support this use case, if you provide a `chain_spec_path` in plain text we use that one to make `our` customizations (that not include override the runtime code) and then we will the `raw` spec using that `chain-spec` file. I have to review if that override the runtime code with the one of the client, if that is the case we should put some logic to prevent that.

Thanks!!

p.d: Did you have a working example to reproduce this behavior.

**Reply by timbrinded** [reply]
Thanks for response, here is my config using local native providers:
```
{
  "settings": {
    "timeout": 1000,
    "provider": "native"
  },
  "relaychain": {
    "chain": "rococo-local",
    "default_command": "tmp/polkadot",
    "default_args": [
      "--no-hardware-benchmarks",
      "-lparachain=debug",
      "--database=paritydb"
    ],
    "nodes": [
      {
        "name": "alice",
        "validator": true
      },
      {
        "name": "bob",
        "validator": true
      }
    ]
  },
  "parachains": [
    {
      "id": 1280,
      "chain": "moonriver-local",
      "chain_spec_path": "tmp/moonriver-plain-spec.json",
      "collator": {
        "name": "alith",
        "command": "../target/release/moonbeam",
        "args": [
          "--no-hardware-benchmarks",
          "-lparachain=debug",
          "--database=paritydb"
        ]
      }
    }
  ],
  "types": {
    "Header": {
      "number": "u64",
      "parent_hash": "Hash",
      "post_state": "Hash"
    }
  }
}
```

Note that `moonriver-plain-spec.json` is generated by using:
```
 ../target/release/moonbeam build-spec --chain moonriver-local > tmp/moonriver-plain-spec.json
```

And overriding genesis.runtime.system.code with the runtime wasm I wish to start the network with

**Comment by timbrinded** [comment]
This was delivered by allowing raw-spec-files to be imported to a zombie node (generated by modifying a plain spec and turning into a raw spec yourself)
